Userland/Libraries/LibJS/Runtime/FinalizationRegistryPrototype.cpp

```diff
@@ -38,4 +43,68 @@ FinalizationRegistry* FinalizationRegistryPrototype::typed_this(VM& vm, GlobalOb
     return static_cast<FinalizationRegistry*>(this_object);
 }
 
+// @STAGE 2@ FinalizationRegistry.prototype.cleanupSome ( [ callback ] ), https://github.com/tc39/proposal-cleanup-some/blob/master/spec/finalization-registry.html
+JS_DEFINE_NATIVE_FUNCTION(FinalizationRegistryPrototype::cleanup_some)
+{
+    auto* finalization_registry = typed_this(vm, global_object);
+    if (!finalization_registry)
+        return {};
+
+    auto callback = vm.argument(0);
+    if (vm.argument_count() > 0 && !callback.is_function()) {
+        vm.throw_exception<TypeError>(global_object, ErrorType::NotAFunction, callback.to_string_without_side_effects());
+        return {};
+    }
+
+    finalization_registry->cleanup(&callback.as_function());
```

Comment:
This will blow up if `callback` is not present, i.e. undefined.

Reply:
```suggestion
    finalization_registry->cleanup(callback.is_undefined() ? nullptr : &callback.as_function());
```

Reply:
my bad, fixed!